frame/lending/src/tests.rs

```diff
@@ -1181,6 +1181,62 @@ fn zero_amount_collateral_deposit() {
 	})
 }
 
+#[test]
+fn halborn_hal03_issue_market_owner_profits_by_liquidate() {
+	// NOTE: As this test shows that changing collateral_factor
+	// can create some loans ready to be liquidated. And similarly changing
+	// other parameters for lending market can create more risks.
+	// So it is recommended to have update_market to go through governance.
+	new_test_ext().execute_with(|| {
+		let (market_id, vault) = create_simple_market();
+
+		let collateral_amount = BTC::units(100);
+		assert_ok!(Tokens::mint_into(BTC::ID, &ALICE, collateral_amount));
+
+		assert_extrinsic_event::<Runtime>(
+			Lending::deposit_collateral(Origin::signed(*ALICE), market_id, collateral_amount),
+			Event::Lending(crate::Event::CollateralDeposited {
+				sender: *ALICE,
+				amount: collateral_amount,
+				market_id,
+			}),
+		);
+
+		let borrow_asset_deposit = USDT::units(1_000_000_000);
+		assert_ok!(Tokens::mint_into(USDT::ID, &CHARLIE, borrow_asset_deposit));
+		assert_ok!(Vault::deposit(Origin::signed(*CHARLIE), vault, borrow_asset_deposit));
+
+		process_and_progress_blocks(1);
+
+		let limit_normalized = Lending::get_borrow_limit(&market_id, &ALICE).unwrap();
+		process_and_progress_blocks(2);
+
+		assert_extrinsic_event::<Runtime>(
+			Lending::borrow(Origin::signed(*ALICE), market_id, limit_normalized),
+			Event::Lending(crate::Event::Borrowed {
+				sender: *ALICE,
+				amount: limit_normalized,
+				market_id,
+			}),
+		);
+
+		assert_eq!(Lending::should_liquidate(&market_id, &ALICE), Ok(false));
+
+		// Update collateral factor to big value
+		let collateral_factor = MoreThanOneFixedU128::saturating_from_rational(2000_u128, 99_u128);
+		let updatable = UpdateInput {
+			collateral_factor,
+			under_collateralized_warn_percent: Percent::from_float(1.1),
+			liquidators: vec![],
+			interest_rate_model: InterestRateModel::default(),
+		};
+		let updatable = updatable.try_into_validated::<UpdateInputVaild>().unwrap();
+		assert_ok!(Lending::update_market(Origin::signed(*ALICE), market_id, updatable));
+		// ALICE loan must be liquidated now.
```

Comment:
Just to understand, Why would ALICE `update_market`/collateral factor to get their own loan liquidated?

Reply:
Sorry for confusion, ALICE is creator of the market so she can update the configs. This is just to show problematic case.

Reply:
Updated test to avoid confusion.